### PR TITLE
Feat/add custom blocks

### LINKED ISSUE #12: Newsletter and White paper widget
Create two widgets - White paper and Newsletter

Newsletter:
- A reusable block element that can be added  multiple times trough the piranha manager page.
- Has an email field and a subscribe button.
- Validation of the email field.
- On click the subscribe button saves the entered email in the db trough a JavaScript post call.
- Upon success/failure a bootstrap modal pops up and displays the status of the post call.

White paper:
 - A reusable block element that can be added  multiple times trough the piranha manager page.
 - Has a single "White paper" button, visible to all and an URL filed visible only trough the piranha manager page.
- "White paper" button:
- - On click a bootstrap modal form pops up.
- - Тhe form has the following fields: Name, Company, Email, Phone, Position.
- - All of the fields require validation.
- - The form has a "Submit" and a "Close" button. 
- - If all of the fields are valid the "Submit" button is active.
- - On click the "Submit" button saves the form data to the db trough a JavaScript post call.
- - Upon failure of the submit action, an error is displayed.
- - Upon success of the submit action, the modal pop up closes and the user is redirected to a URL the is provided trough the piranha manager page.
- - On click the "Close" button clears the data and closes the form.
- URL field:
- - A field visible only in the piranha manager page.
- - Upon entering the field and saving the block element the entered URL is saved to the db.
- - The saved URL is used for redirecting the user upon successfully submitting the "White paper" form.

### PR DESCRIPTION
**Pull request by mkyordanov**
Added custom blocks for white paper and newsletter
Added google recaptcha

### CODE REVIEW

File: Views/Cms/DisplayTemplates/NewsletterViewModel.cshtml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            <div class="modal-body" id="newsletterReturnStatus">
+            </div>
+            <div class="modal-footer">
+                <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
+            </div>
+        </div>
+    </div>
+</div>
+
+<script>
```
Comment (adautev) [inline Views/Cms/DisplayTemplates/NewsletterViewModel.cshtml]: Scale this out to a separate newsletter.js in assets. 
Make sure newsletter.js is built into the main javascript file and obfuscated for a production build.

File: Views/Cms/DisplayTemplates/WhitePaperViewModel.cshtml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            <div class="modal-footer">
+                <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
+                <button type="button" class="btn btn-primary" onclick="submitWhitePaper()">Submit</button>
+            </div>
+        </div>
+    </div>
+</div>
+
+
+<script>
```
Comment (adautev) [inline Views/Cms/DisplayTemplates/WhitePaperViewModel.cshtml]: Scale this out to a separate Whitepaper.js in assets. 
Make sure newsletter.js is built into the main javascript file and obfuscated for a production build.

File: Views/Shared/_Layout.cshtml [hunk 3]
```diff
@@ -29,20 +33,22 @@
...
     </div>
 </nav>
 
-@RenderBody()
-
-<script src="https://code.jquery.com/jquery-3.4.1.slim.min.js" integrity="sha384-J6qa4849blE2+poT4WnyKhv5vZF5SrPo0iEjwBvKU7imGFAV0wwj1yYfoRSJoZ+n" crossorigin="anonymous"></script>
+<script src="https://www.google.com/recaptcha/api.js?render=@GoogleRecaptchaConfig.Value.SiteKey"></script>
+<script src="https://code.jquery.com/jquery-3.6.0.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>
 <script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js" integrity="sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo" crossorigin="anonymous"></script>
 <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/js/bootstrap.min.js" integrity="sha384-wfSDF2E50Y2D1uUdj0O3uMBJnjuUD4Ih7YwaYd1iqfktj0Uod8GCExl3Og8ifwB6" crossorigin="anonymous"></script>
```
Comment (adautev) [inline Views/Shared/_Layout.cshtml]: Bootstrap version should be latest. Make sure js and css bootstrap assets are latest.

File: appsettings.Development.json [hunk 4]
```diff
@@ -10,5 +10,9 @@
   "ConnectionStrings": {
     "piranha": "Filename=./piranha.db",
     "blobstorage": "DefaultEndpointsProtocol=https;AccountName=;AccountKey=;EndpointSuffix="
+  },
+  "GoogleReCaptcha": {
```
Comment (adautev) [inline appsettings.Development.json:14]: Submitting this in a public repo is highly not recommended. Please:

- Reset the keys in google admin.
- Check in a .json file with empty keys
- Don't forget every time when you change appsettings to commit a model appsetting containing the changes, but without sensitive information.
- Add the model appsettings*.json to .gitignore

Comment (mkyordanov) [reply appsettings.Development.json:14]: Done

File: Views/Shared/_Layout.cshtml (hunk 3 above)
Comment (mkyordanov) [reply Views/Shared/_Layout.cshtml]: Done

File: Views/Cms/DisplayTemplates/WhitePaperViewModel.cshtml (hunk 2 above)
Comment (mkyordanov) [reply Views/Cms/DisplayTemplates/WhitePaperViewModel.cshtml]: Done

File: Views/Cms/DisplayTemplates/NewsletterViewModel.cshtml (hunk 1 above)
Comment (mkyordanov) [reply Views/Cms/DisplayTemplates/NewsletterViewModel.cshtml]: Done

**Pull request merged**